QoLPack/jars/src/data/plugins/qolp_modPlugin.java

```diff
@@ -12,17 +12,84 @@
     public static final String ID = "qolp_modPlugin";
     public static final String SETTINGS_PATH = "QoLPack.ini";
 
+    private JSONObject settings;
+
+    @Override
+    public void afterGameSave() {
+        if (hasQol("ScavengeAsYouFly")) {
+            restoreAutoScavenge();
+        }
+    }
+
+    @Override
+    public void beforeGameSave() {
+        if (hasQol("ScavengeAsYouFly")) {
+            removeAutoScavenge();
+        }
+    }
+
+    @Override
+    public void onApplicationLoad() throws Exception {
+        settings = Global.getSettings().getMergedJSONForMod(SETTINGS_PATH, ID);
+        if (hasQol("BetterSensorBurst")) {
+            Global.getSettings().getAbilitySpec("sensor_burst").getTags().remove("burn-");
+            Global.getSettings().getAbilitySpec("sensor_burst").getTags().remove("sensors+");
+        }
+    }
+
     @Override
     public void onGameLoad(boolean newGame) {
-        boolean clock = true;
-        try {
-            JSONObject cfg = Global.getSettings().getMergedJSONForMod(SETTINGS_PATH, ID);
-            clock = cfg.getBoolean("EnableClock");
-        } catch (IOException | JSONException e) {
-            e.printStackTrace();
+        if (hasQol("EnableClock")) {
+            addTransientScript(new qolp_clock());
+        }
+        if (hasQol("PartialSurveyAsYouFly")) {
+            addTransientScript(new qolp_PartialSurveyScript());
         }
-        if (clock && !Global.getSector().hasScript(qolp_clock.class)) {
-            Global.getSector().addTransientScript(new qolp_clock());
+        if (hasQol("ScavengeAsYouFly")) {
+            restoreAutoScavenge();
+            notifyAboutState();
         }
+        if (hasQol("Transpoffder")) {
+            addTransientListener(new qolp_TranspoffderListener());
+        }
+    }
+
+    private void addTransientListener(Object listener) {
+        Global.getSector().getListenerManager().addListener(listener, true);
+    }
+
+    private void addTransientScript(EveryFrameScript script) {
+        Global.getSector().addTransientScript(script);
+    }
+
+    private boolean hasQol(String key) {
+        return settings.optBoolean(key, true);
+    }
+
+    private void notifyAboutState() {
+        String state = AutoScavengeAbility.isOn() ? "enabled" : "disabled";
+        Global
+            .getSector()
+            .getCampaignUI()
+            .addMessage(
+                "Automatic scavenging is %s.",
+                Misc.getTextColor(),
+                state,
+                state,
+                Misc.getHighlightColor(),
+                Misc.getHighlightColor()
+            );
+    }
+
+    private void removeAutoScavenge() {
+        boolean isOn = AutoScavengeAbility.isOn();
+        Global.getSector().getMemoryWithoutUpdate().set("$transpoffderAutoScavenge", isOn);
```

Comment:
You may want to keep this memkey for compatibility with  / switchover from `Transponder Off` mod. Not a huge deal, might as well rename it to `qolp_AutoScavenge`.